app/models/queen.rb

```diff
@@ -1,8 +1,9 @@
 class Queen < Piece
 
-    def valid_move?(new_x, new_y)
-        return false if square_occupied?(x_position, y_position)
-        return false if self.is_obstructed?(x_position, y_position)
-        horizontal_move?(x_position, y_position) ||  vertical_move?(x_position, y_position) ||  diagonal_move?(x_position, y_position)
-    end
+  def valid_queen_move(new_x, new_y)
+    vertical_move = self.x_position === new_x && self.y_position != new_y
+    horizontal_move = self.y_position === new_y && self.x_position != new_x
+    diagonal_move = 
```

Comment:
Looking forward for diagonal 😄

Reply:
Just a suggestion of how this might be able to be refactored and have all the moves accounted for:

`class Queen < Piece
       def valid_queen_move?(new_x, new_y)
            unless is_obstructed?(new_x.to_i, new_y.to_i)
                 return true if 
  #diagonal                    (new_x.to_i - x_position).abs == (new_y.to_i - y_position).abs || 
  #vertical                   new_x.to_i == x_position || 
  #horizontal                     new_y.to_i == y_position
            end
            false
     end
end`